src/edu/csus/ecs/pc2/clics/API202306/GroupService.java

```diff
@@ -0,0 +1,120 @@
+ * @author John Buck
+ *
+ */
+@Path("/contests/{contestId}/groups")
+@Produces(MediaType.APPLICATION_JSON)
+@Provider
+@Singleton
+public class GroupService implements Feature {
+
+    private IInternalContest model;
+    @SuppressWarnings("unused")
+    private IInternalController controller;
+
+    public GroupService(IInternalContest inContest, IInternalController inController) {
+        super();
+        this.model = inContest;
+        this.controller = inController;
+    }
+
+    /**
+     * This method returns a representation of the current contest groups in JSON format.
+     * The response is a JSON array with one group description per array element, complying with 2023-06
+     *
+     * @param contestId The contest id
+     * @return a {@link Response} object containing the groups in JSON form
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getGroups(@PathParam("contestId") String contestId) {
+
+        // check contest id
+        if(contestId.equals(model.getContestIdentifier()) == false) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+
+        ArrayList<CLICSGroup> glist = new ArrayList<CLICSGroup>();
+
+        for(Group group: model.getGroups()) {
+            if (group.isDisplayOnScoreboard()) {
+                glist.add(new CLICSGroup(group));
+            }
+        }
+        try {
+            ObjectMapper mapper = JSONUtilities.getObjectMapper();
+            String json = mapper.writeValueAsString(glist);
+            return Response.ok(json, MediaType.APPLICATION_JSON).build();
+        } catch (Exception e) {
+            return Response.status(Status.INTERNAL_SERVER_ERROR).entity("Error creating JSON for groups " + e.getMessage()).build();
+        }
+    }
+
+
+    /**
+     * Returns a representation of the specified groupid in the specified contestid.
+     * The response is a JSON object describing the group, complying with 2023-06
+     *
+     * @param contestId The contest
+     * @param groupId The desired group in the contest
+     * @return a {@link Response} object containing the groups in JSON form
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("{groupId}/")
+    public Response getGroup(@PathParam("contestId") String contestId, @PathParam("groupId") String groupId) {
+
+        // check contest id
+        if(contestId.equals(model.getContestIdentifier()) == true) {
+            for(Group group: model.getGroups()) {
+                if (group.isDisplayOnScoreboard() && IJSONTool.getGroupId(group).equals(groupId)) {
+                    return Response.ok(new CLICSGroup(group).toJSON(), MediaType.APPLICATION_JSON).build();
+                }
+            }
+        }
+        return Response.status(Response.Status.NOT_FOUND).build();
+    }
+
+    /**
+     * Retrieve access information about this endpoint for the supplied user's security context
+     *
+     * @param sc User's security information
```

Comment:
Unused parameter "contest" should be documented.

Reply:
Fixed in next push (comment added)